src/CoreWCF.WebHttp/src/CoreWCF/UriTemplate.cs

```diff
@@ -0,0 +1,1799 @@
+                            throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(
+                                SR.Format(SR.UTDefaultValueToQueryVarFromAdditionalDefaults, _originalTemplate,
+                                uppercaseKey)));
+                        }
+                        else if (string.Compare(kvp.Value, UriTemplate.NullableDefault, StringComparison.OrdinalIgnoreCase) == 0)
+                        {
+                            throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(
+                                SR.Format(SR.UTNullableDefaultAtAdditionalDefaults, _originalTemplate,
+                                uppercaseKey)));
+                        }
+                        else
+                        {
+                            if (_additionalDefaults == null)
+                            {
+                                _additionalDefaults = new Dictionary<string, string>(additionalDefaults.Count, StringComparer.OrdinalIgnoreCase);
+                            }
+                            _additionalDefaults.Add(kvp.Key, kvp.Value);
+                        }
+                    }
+                }
+            }
+
+            // Validate defaults (if should)
+            if ((variables != null) && (variables.DefaultValues != null))
+            {
+                variables.ValidateDefaults(out _firstOptionalSegment);
+            }
+            else
+            {
+                _firstOptionalSegment = _segments.Count;
+            }
+        }
+
+        public IDictionary<string, string> Defaults
+        {
+            get
+            {
+                if (_defaults == null)
+                {
+                    Interlocked.CompareExchange(ref _defaults, new UriTemplateDefaults(this), null);
+                }
+                return _defaults;
+            }
+        }
+
+        public bool IgnoreTrailingSlash { get; }
+
+        public ReadOnlyCollection<string> PathSegmentVariableNames
+        {
+            get
+            {
+                if (variables == null)
+                {
+                    return VariablesCollection.EmptyCollection;
+                }
+                else
+                {
+                    return variables.PathSegmentVariableNames;
+                }
+            }
+        }
+        public ReadOnlyCollection<string> QueryValueVariableNames
+        {
+            get
+            {
+                if (variables == null)
+                {
+                    return VariablesCollection.EmptyCollection;
+                }
+                else
+                {
+                    return variables.QueryValueVariableNames;
+                }
+            }
+        }
+
+        internal bool HasNoVariables
+        {
+            get
+            {
+                return variables == null;
+            }
+        }
+        internal bool HasWildcard
+        {
```

Comment:
Formatting, missing a blank line between methods

Reply:
This is addressed in e95ff7c922db5d4a6ee55847b763b49619f08311.